appinventor/components/src/com/google/appinventor/components/runtime/ButtonBase.java

```diff
@@ -390,12 +397,20 @@ private void updateAppearance() {
     }
   }
 
+  private ColorStateList createRippleState () {
+
+    int[][] states = new int[][] { new int[] { android.R.attr.state_enabled} };
+    int enabled_color = defaultColorStateList.getColorForState(view.getDrawableState(), android.R.attr.state_enabled);
+    int[] colors = new int[] { Color.argb(70, Color.red(enabled_color), Color.green(enabled_color),
+            Color.blue(enabled_color)) };
+
+    return new ColorStateList(states, colors);
+  }
+
   // Throw IllegalArgumentException if shape has illegal value.
   private void setShape() {
     ShapeDrawable drawable = new ShapeDrawable();
-    // Set color of drawable.
```

Comment:
I think this is the part of the change causing the bad default color behavior with shaped buttons.

Reply:
OK, yes. The color is set with ColorFilter below. I missed the check for default when I moved it. Thanks.